src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -124,6 +128,23 @@ public OtlpExportProtocol Protocol
         set => this.protocol = value;
     }
 
+    /// <summary>
+    /// Gets or sets the user agent identifier.
+    /// </summary>
+    public string UserAgentProductIdentifier
+    {
+        get => this.userAgentProductIdentifier ?? string.Empty;
+        set
+        {
+            this.userAgentProductIdentifier = string.IsNullOrWhiteSpace(value) ? string.Empty : value;
+
+            standardHeaders =
+            [
+                new("User-Agent", this.GetUserAgentString())
+            ];
+        }
```

Comment:
This is weird to me. The property changes state, that then affects the behaviour of `GetUserAgentString()`, then sets another piece of state.

I feel there's a better way to write this that doesn't depend on the order of operations.